src/chainstate/burn/sortition.rs

```diff
@@ -135,30 +131,24 @@ impl BlockSnapshot {
     /// Note that the VRF seed is not guaranteed to be the hash of a valid VRF
     /// proof.  Miners would only build off of leader block commits for which they
     /// (1) have the associated block data and (2) the proof in that block is valid.
-    fn select_winning_block<'a, A, K>(tx: &mut Transaction<'a>, block_height: u64, sortition_hash: &SortitionHash, burn_dist: &Vec<BurnSamplePoint<A, K>>) -> Result<Option<LeaderBlockCommitOp<A, K>>, db_error>
-    where
-        A: Address,
-        K: PublicKey
-    {
-        // get the last winner's VRF seed
-        let last_sortition_snapshot = BurnDB::<A, K>::get_last_snapshot_with_sortition(tx, block_height)?;
-        let VRF_seed;
-
-        if last_sortition_snapshot.sortition_hash == SortitionHash::initial() {
-            // this is the sentinal "first-sortition" block 
-            VRF_seed = VRFSeed::initial();
-        }
-        else {
-            // there was a prior winning block commit.  Use its VRF seed.
-            let last_winning_block_commit_opt = BurnDB::<A, K>::get_block_commit(tx, &last_sortition_snapshot.winning_block_txid, &last_sortition_snapshot.winning_block_burn_hash)?;
-            if last_winning_block_commit_opt.is_none() {
-                // something is seriously wrong -- the canonical block snapshot doesn't point to a
-                // canonical block commit.
-                panic!("FATAL ERROR: no block commit for snapshot {:?}", last_sortition_snapshot);
+    fn select_winning_block<'a>(tx: &mut BurnDBTx<'a>, block_header: &BurnchainBlockHeader, sortition_hash: &SortitionHash, burn_dist: &Vec<BurnSamplePoint>) -> Result<Option<LeaderBlockCommitOp>, db_error> {
+        let burn_block_height = block_header.block_height;
+
+        // get the last winner's VRF seed in this block's fork
+        let last_sortition_snapshot = BurnDB::get_last_snapshot_with_sortition(tx, burn_block_height - 1, &block_header.parent_block_hash)
+            .expect("FATAL: failed to query last snapshot with sortition");
+
+        let VRF_seed =
+            if last_sortition_snapshot.sortition_hash == SortitionHash::initial() {
```

Comment:
Should this if condition call `is_initial()`?

Reply:
It shouldn't make a difference -- absent a hash collision, there's no way to get a sortition hash whose bytes are all 0's.  I don't think the `is_initial()` method is even used anywhere.

Reply:
Right -- we should either delete `is_initial()` because its unused, or use it to avoid duplication of logic

Reply:
Fixed in 8f8c6e10